src/roiextractors/extractors/femtonicsimagingextractor/femtonicsimagingextractor.py

```diff
@@ -0,0 +1,418 @@
+        return "".join(chr(x) for x in arr if x != 0)
+
+    def get_image_shape(self) -> tuple[int, int]:
+        """Get the shape of the video frame (num_rows, num_columns)."""
+        return self._video.shape[1], self._video.shape[2]  # (height, width)
+
+    def get_num_samples(self) -> int:
+        """Get the number of samples (frames) in the video."""
+        return self._video.shape[0]
+
+    def get_channel_names(self) -> list[str]:
+        """Get the channel names."""
+        return [self._selected_channel_name]
+
+    def get_sampling_frequency(self) -> float:
+        """Get the sampling frequency in Hz."""
+        return self._sampling_frequency
+
+    def _get_frame_duration_in_milliseconds(self) -> Optional[float]:
+        """Get time per frame from metadata."""
+        attrs = dict(self._file_handle[self._session_key][self._munit_key].attrs)
+        return attrs.get("ZAxisConversionConversionLinearScale")
+
+    # Femtonics-specific getter methods
+    def _get_session_uuid(self):
+        """
+        Get the session UUID from the root attributes.
+
+        Returns
+        -------
+        uuid : array or value
+            The session UUID.
+        """
+        attrs = dict(self._file_handle.attrs)
+        return attrs.get("Uuid")
+
+    def _get_pixel_size_in_micrometers(self) -> tuple[float, float]:
+        """Get pixel size in micrometers."""
+        attrs = dict(self._file_handle[self._session_key][self._munit_key].attrs)
+
+        x_size = attrs.get("XAxisConversionConversionLinearScale", 1.0)
+        y_size = attrs.get("YAxisConversionConversionLinearScale", 1.0)
+
+        # Check units and warn if not micrometers
+        x_units = self._decode_string(attrs.get("XAxisConversionUnitName", []))
+        y_units = self._decode_string(attrs.get("YAxisConversionUnitName", []))
+
+        def is_micrometers(unit_str):
+            if not unit_str:
+                warn("No X/Y axis unit specified in metadata; assuming micrometers (μm).")
+                return True  # Assume micrometers if no unit specified
+            unit_lower = unit_str.lower()
+            return any(um_variant in unit_lower for um_variant in ["μm", "µm", "um", "microm", "micro", "micrometer"])
+
+        if x_units and not is_micrometers(x_units):
+            warn(f"X-axis units are '{x_units}', expected micrometers")
+        if y_units and not is_micrometers(y_units):
+            warn(f"Y-axis units are '{y_units}', expected micrometers")
+
+        return x_size, y_size
+
+    def _get_image_shape_metadata(self) -> tuple[int, int, int]:
+        """
+        Get the image shape metadata (X, Y, Z dimensions) from the measurement unit attributes.
+
+        Returns
+        -------
+        shape : tuple
+            (XDim, YDim, ZDim) as (num_columns, num_rows, num_frames)
+        """
+        munit_attrs = dict(self._file_handle[self._session_key][self._munit_key].attrs)
+        x_dim = munit_attrs.get("XDim")
```

Comment:
```suggestion
        x_dim = int(munit_attrs.get("XDim"))
```
Because otherwise you get numpy scalars:

https://numpy.org/doc/stable/reference/arrays.scalars.html

And that is why you test is failing.